src/Stormpath/Resource/FacebookProvider.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Stormpath\Resource;
+
+/*
+ * Copyright 2013 Stormpath, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use Stormpath\Client;
+use Stormpath\DataStore\InternalDataStore;
+use Stormpath\Stormpath;
+
+class FacebookProvider extends Provider
+{
+    const CLIENT_ID             = 'clientId';
+    const CLIENT_SECRET         = 'clientSecret';
+    const REDIRECT_URI          = 'redirectUri';
+    const FACEBOOK_PROVIDER_ID  = 'facebook';
+
+    public static function get($href, array $options = array())
+    {
+        if (substr($href, 0 - strlen(self::PATH)) != self::PATH)
+        {
+            $href = $href.'/'.self::PATH;
+        }
+
+        if (!(stripos($href, 'http') === 0) and !(stripos($href, Directory::PATH) === 0))
+        {
+            $href = Directory::PATH.'/'.$href;
+        }
+
+        return Client::get($href, Stormpath::FACEBOOK_PROVIDER, null, $options);
```

Comment:
I think this call should work if you just make sure that href has "/provider" after the id and instead of null for the path argument, you could pass `Directory::PATH`. Same applies for `GoogleProvider`.